index.rst

```diff
@@ -1,52 +1,163 @@
+
+PSFex fits PSF variations simultaneously with the PSF itself. Variability may be taken with respect to source position, source parameters, or image metadata (while PSFex itself requires that this input be in SExtractor and FITS format, respectively, ``meas_extensions_psfex`` does the appropriate translation from Sources and Exposures). PSF variations are represented by a low-order polynomial whose "coefficients" are themselves PSF images. There is no support for high-frequency components of the type discussed by [CMJ2012]_. The algorithm requires :math:`\mathcal{O}((N_\mathrm{PSF} N_\mathrm{coeff})^3 + N_\mathrm{PSF}^2 N_\mathrm{star} N_\mathrm{coeff})` time, where :math:`N_\mathrm{coeff}` is the number of terms in the polynomial.
+
+PSFEx supports the use of PCA to get an optimized image basis. Presumably this is in the context of PSF variation fitting, but the documentation is very vague about what the principal component analysis produces. It can also produce PSF homogenization kernels on request.
+
+.. _psffit-general-piff:
+
+PIFF
+----
+
+`PIFF <http://rmjarvis.github.io/Piff/html/>`_ is a new PSF fitting tool that is still under development. As most of the current version consists of placeholders (e.g., the only PSF model supported is a single elliptical Gaussian), I will discuss this program only briefly.
+
+PIFF is designed for wide-field telescopes and multi-chip cameras. It plans to offer fitting of PSFs as a sum of basis functions, which may be pixel/nonparametric, shapelet, or Gaussian. PSF variations will be supported (only?) with respect to position, and may be interpolated using polynomials, kriging, or :ref:`PSFEnt <psffit-atmo-psfent>`. It will also allow the user to specify the optical component of the PSF while fitting for the atmospheric contribution.
+
+It is not clear how PIFF will select PSF template sources, except that it will involve some kind of outlier rejection scheme.
+
+.. _psffit-dense:
+
+PSF Fitting in Dense Fields
+===========================
+
+Iterative Derivation
+--------------------
+
+[ABP2006]_ present a method to fit the PSF in dense fields for WFI on the ESO 2.2m. Beginning with crude source detections, centroided positions, and aperture fluxes, they reconstruct the PSF from visible stars, use the PSF to get new source positions and fluxes, use the improved source data to construct better PSFs, and so on. Template stars are selected to have high counts and no nearby neighbors; stars that are poor fits to the PSF can be rejected.
+The PSF is nonparametric, but with smoothness and centering constraints enforced at each iteration.
+
+Each iteration of the algorithm requires :math:`\mathcal{O}(N_\mathrm{PSF} N_\mathrm{star} + N_\mathrm{PSF} N_\mathrm{smooth})` time, where :math:`N_\mathrm{smooth}` is the number of pixels in the PSF smoothing kernel. No stopping condition is provided, although the discussion suggests the algorithm converges quickly.
```

Comment:
There must be some stopping condition internally, right?  Otherwise wouldn't it just bobble around at the machine precision limit?